### PR TITLE
chore: reorganizing primitive exports

### PR DESCRIPTION
**Pull request by zchenwei**
*Issues:*

```
interface UseAmplifyOutput {
  components: React.ReactNode;
  theme: Theme;
}
```

Typing components roughly with `React.ReactNode` will confuse Typescript for a ReactNode is not necessarily callable.

Screen Shot 2021-11-11 at 3 29 57 PM

The workaround here is to reorganize the existing exports and separate out the component-only exports from primitives package for explicit typing purpose and this way also provides us with intellisense hints 

*Description of changes:*

Reorganize primitive exports and update the return type of `useAmplify` hook. 

Intellisense hints:

Screen Shot 2021-11-11 at 4 27 27 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/react/src/primitives/primitives.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+export { Alert } from './Alert';
```
Comment (reesscot) [inline packages/react/src/primitives/primitives.ts:1]: Is this change necessary, now that our exports are all named exports?

File: packages/react/src/hooks/useAmplify.ts [hunk 2]
```diff
@@ -3,10 +3,10 @@ import * as React from 'react';
 import { Theme } from '@aws-amplify/ui';
 
 import { AmplifyContext } from '../components/AmplifyProvider/AmplifyContext';
-import * as primitives from '../primitives';
+import * as primitives from '../primitives/primitives';
```
Comment (reesscot) [inline packages/react/src/hooks/useAmplify.ts:6]: Feels awkward to export from `primitives/primitives`

**Review by reesscot — CHANGES_REQUESTED** [review]
Can you rebase from main and see if the index -> primitives file change is necessary?

File: packages/react/src/primitives/primitives.ts (hunk 1 above)
Comment (zchenwei) [reply packages/react/src/primitives/primitives.ts:1]: I don't see the relation with named exports. There will be hooks exported  which we don't want to include in primitives?

File: packages/react/src/hooks/useAmplify.ts (hunk 2 above)
Comment (zchenwei) [reply packages/react/src/hooks/useAmplify.ts:6]: Rename to `primitives/components.ts`

Comment (reesscot) [reply packages/react/src/hooks/useAmplify.ts]: This file name feels better

File: packages/react/src/primitives/primitives.ts (hunk 1 above)
Comment (reesscot) [reply packages/react/src/primitives/primitives.ts]: OK, I understand this a bit better. Still feels awkward, but it makes sense.

**Review by reesscot — APPROVED** [review]
